### PR TITLE
[Transform] Improve force stop robustness in case of an error

### PR DESCRIPTION
**Pull request by hendrikmuhs**
If a transform config got lost (e.g. because the internal index disappeared) tasks could not be stopped using transform API. This change makes it possible to stop transforms without a config, meaning to remove the background task. In order to do so force must be set to true.

### CODE REVIEW

**Review by dimitris-athanasiou — COMMENTED** [review]
Looks good. A few minor things to go through.

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportStopTransformAction.java [hunk 1]
```diff
@@ -127,6 +134,25 @@ static void validateTaskState(ClusterState state, List<String> transformIds, boo
...
 
+    static Tuple<Set<String>, Set<String>> findTasksWithoutConfig(ClusterState state, String transformId) {
+        PersistentTasksCustomMetaData tasks = state.metaData().custom(PersistentTasksCustomMetaData.TYPE);
+
+        Set<String> taskIds = new HashSet<>();
+        Set<String> executorNodes = new HashSet<>();
+
+        Predicate<PersistentTask<?>> taskMatcher = Strings.isAllOrWildcard(new String[] { transformId }) ? t -> true : t -> {
+            TransformTaskParams transformParams = (TransformTaskParams) t.getParams();
+            return transformParams.getId().equals(transformId);
```
[the remark this answers is not in the document]
Comment (hendrikmuhs) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportStopTransformAction.java]: yes, that was my intention. In this situation I think you either have 1 transform that is broken or you want to rest everything. Supporting such patterns makes it very complicated and its dangerous, too.

However, I will add a check and throw an exception if someone tries to call it with "my_id*".

[the remark this answers is not in the document]
Comment (hendrikmuhs) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportStopTransformAction.java]: hmm, this gets complicated. I changed my mind and will allow it.

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM Cool, the only thing I'd add would be a test for when the tranform id in the _stop call is a wildcard pattern (e.g. "foo-*"). But i'll leave it on you whether to add it in this PR or another.

**Pull request merged**